plugins/modules/api.py

```diff
@@ -73,118 +136,76 @@
         username: "{{ username }}"
-        path: "{{ path }}"
+        path: "ip address"
         query: ".id address WHERE address == {{ ip2 }}"
-      register: queryout
-
-    - name: Dump "Query for" output and set fact with ".id" for "{{ ip2 }}"
-      ansible.builtin.debug:
-        msg: '{{ queryout }}'
 
-    - name: Store query_id for later usage
-      ansible.builtin.set_fact:
-        query_id: "{{ queryout['msg'][0]['.id'] }}"
-
-    - name: Update ".id = {{ query_id }}" taken with custom fact "fquery_id"
+    - name: Extended query example - ".id,address,network" where address is not 192.168.255.10/24 or is 10.20.36.20/24
       community.routeros.api:
         hostname: "{{ hostname }}"
         password: "{{ password }}"
         username: "{{ username }}"
-        path: "{{ path }}"
-        update: >-
-            .id={{ query_id }}
-            address={{ ip3 }}
-            comment={{ 'A comment with spaces' | community.routeros.quote_argument_value }}
-      register: updateout
-
-    - name: Dump "Update" output
-      ansible.builtin.debug:
-        msg: '{{ updateout }}'
-
-    - name: Remove ips - stage 1 - query ".id" for "{{ ip2 }}" and "{{ ip3 }}"
+        path: "ip address"
+        extended_query:
+          attributes:
+            - network
+            - address
+            - .id
+          where:
+            - attribute: "network"
+              is: "=="
+              value: "192.168.255.0"
+            - or:
+                - attribute: "address"
+                  is: "!="
+                  value: "192.168.255.10/24"
+                - attribute: "address"
+                  is: "eq"
+                  value: "10.20.36.20/24"
+            - attribute: "network"
+              is: "in"
+              value:
+                 - "10.20.36.0"
+                 - "192.168.255.0"
+
+    - name: Update example - ether2 ip addres with ".id = *14"
       community.routeros.api:
         hostname: "{{ hostname }}"
         password: "{{ password }}"
         username: "{{ username }}"
-        path: "{{ path }}"
-        query: ".id address WHERE address == {{ item }}"
-      register: id_to_remove
-      loop:
-        - "{{ ip2 }}"
-        - "{{ ip3 }}"
-
-    - name: Set fact for ".id" from "Remove ips - stage 1 - query"
-      ansible.builtin.set_fact:
-        to_be_remove: "{{ to_be_remove |default([]) + [item['msg'][0]['.id']] }}"
-      loop: "{{ id_to_remove.results }}"
-
-    - name: Dump "Remove ips - stage 1 - query" output
-      ansible.builtin.debug:
-        msg: '{{ to_be_remove }}'
-
-    # Remove "{{ rmips }}" with ".id" by "to_be_remove" from query
-    - name: Remove ips - stage 2 - remove "{{ ip2 }}" and "{{ ip3 }}" by '.id'
+        path: "ip address"
+        update: >-
+            .id=*14
+            address=192.168.255.20/24
+            comment={{ 'Update 192.168.255.10/24 with .id=*14 to 192.168.255.20/24 on ether2' | community.routeros.quote_argument_value }}
```

Comment:
Hardcoding temporary IDs in comments is not a good idea :)
```suggestion
            comment={{ 'Update 192.168.255.10/24 to 192.168.255.20/24 on ether2' | community.routeros.quote_argument_value }}
```